code/hud/hudbrackets.cpp

```diff
@@ -395,99 +395,100 @@ void HudGaugeBrackets::renderObjectBrackets(object *targetp, color *clr, int w_c
 	int bound_rc;
 	SCP_list<CJumpNode>::iterator jnp;
 
-	if ( Player->target_is_dying <= 0 ) {
-		int modelnum;
-
-		switch ( targetp->type ) {
-		case OBJ_SHIP:
-			modelnum = Ship_info[Ships[targetp->instance].ship_info_index].model_num;
-			bound_rc = model_find_2d_bound_min( modelnum, &targetp->orient, &targetp->pos,&x1,&y1,&x2,&y2 );
-			if ( bound_rc != 0 ) {
-				draw_box = false;
-			}
-			break;
-
-		case OBJ_DEBRIS:
-			modelnum = Debris[targetp->instance].model_num;
-			bound_rc = submodel_find_2d_bound_min( modelnum, Debris[targetp->instance].submodel_num, &targetp->orient, &targetp->pos,&x1,&y1,&x2,&y2 );
-			if ( bound_rc != 0 ) {
-				draw_box = false;
-			}
-			break;
-
-		case OBJ_WEAPON:
-			modelnum = Weapon_info[Weapons[targetp->instance].weapon_info_index].model_num;
-			if (modelnum != -1)
-				bound_rc = model_find_2d_bound_min( modelnum, &targetp->orient, &targetp->pos,&x1,&y1,&x2,&y2 );
-			else {
-				vertex vtx;
-				g3_rotate_vertex(&vtx,&targetp->pos);
-				g3_project_vertex(&vtx);
-				x1 = x2 = (int) vtx.screen.xyw.x;
-				y1 = y2 = (int) vtx.screen.xyw.y;
-			}
+	if (Player->target_is_dying > 0 && targetp == &Objects[Player_ai->target_objnum])
```

Comment:
So this is the logic change, right?  It looks like the order that the code is executed has changed?  Not sure what's going on.

Reply:
Yeah its kind of confusing look at it in github. That is correct, the initial similar if is replaced with this which does an early return, the rest of the function was un-indented since it no longer in an if statement.

Reply:
I guess it probably would've been cleaner if just added `|| targetp == &Objects[Player_ai->target_objnum]` into the old `if` ...